src/api/modelTypes.js

```diff
@@ -78,62 +78,25 @@ export type DevUser = $ReadOnly<{|
  * A Zulip user, which might be a human or a bot, as found in one realm.
  *
  * This is a user object as found in properties `realm_users` and
- * `realm_non_active_users` of a `/register` response.
- *
- * For details on the properties, see the Zulip API docs on `/users`:
- *   https://zulip.com/api/get-users#response
- * which returns almost the same set of properties.
- *
- * See also the comments on `UserProfile` in the server (lineno is approx.):
- *   https://github.com/zulip/zulip/blob/main/zerver/models.py#L734
- * Most properties correspond to fields on `UserProfile`, and many are
- * described most usefully there.
- *
- * For authoritative results, consult how `raw_users`, and then
- * `realm_users` and `realm_non_active_users`, are computed in
- * `zulip/zulip:zerver/lib/events.py` .
- *
- * Properties are listed below in the order they appear on `UserProfile`,
- * because that's the most logically-organized and also the most helpful
- * of the references above.
+ * `realm_non_active_users` of a `/register` response. See the API doc:
+ *   https://zulip.com/api/register-queue
  *
  * See also:
  *  * `CrossRealmBot` for another type of bot user, found in a
  *    different part of a `/register` response
  *  * `UserOrBot` for a convenience union of the two
  */
 export type User = $ReadOnly<{|
-  user_id: UserId,
-  email: string,
-
-  full_name: string,
-
-  date_joined: string,
-
-  // is_active doesn't appear in `/register` responses -- instead,
-  // users where is_active is true go in `realm_users`, and where false
-  // go in `realm_non_active_users`.  Shrug.
-
-  // is_admin corresponds to is_realm_admin in server code.
-  is_admin: boolean,
-
-  // is_guest included since commit d5df0377c (in 1.9.0); before that,
-  // there's no such concept, so effectively it's implicitly false.
-  is_guest?: boolean,
+  // Property ordering follows the doc.
+  // Current to feature level 121.
 
   // For background on the "*bot*" fields, see user docs on bots:
   //   https://zulip.com/help/add-a-bot-or-integration
   // Note that certain bots are represented by a different type entirely,
   // namely `CrossRealmBot`.
-  is_bot: boolean,
```

Comment:
nit: Let's keep this comment next to one of the fields it's about, to increase the chance that it's noticed when it's relevant. Perhaps `is_bot` -- that seems like the one where you're most likely to be looking and be in need of the context that `CrossRealmBot` is over there.